### DISCUSSION TITLE
Batch inserts using DGS or WebClientGraphQLClient.reactiveExecuteQuery

### DISCUSSION THREAD

**Discussion by raghuraman-varadharajan** [Netflix/dgs-framework#1176] [Q&A]
Hello,
We have been able to implement most of our GraphQL mutations using the `WebClientGraphQLClient.reactiveExecuteQuery(query, variables)` method. However we are looking to improve limiting the number of mutation queries by doing batching  by inserting a batch of 100 rows per query. 

However, the `variables` in the `WebClientGraphQLClient.reactiveExecuteQuery(query, variables)` is a Map<String, Object> . If I want to bulk insert a list of rows then how would I pass the parameter to the reactiveExecuteQueryMethod for a query like below 

```
        mutation MyMutation($objects: [PersonInput!], $update_columns: [person_update_column!] = []) {
          insert_person(objects: $objects,
            on_conflict: {constraint: person_unique_key, update_columns: $update_columns}) {
            affected_rows
          }
        }
```

Would it be a `Map.of('objects', List.of(new Person(), new Person()))` or should I do it some other way

**Comment by Gajanana** [comment]
It would be something like this 
```
           final Map<String, Object> p1 = new HashMap<>();
            p1.put("name", name1);
            p1.put("id", id1);
            final Map<String, Object> p2 = new HashMap<>();
            p1.put("name", name2);
            p1.put("id", id2);

           Map.of('objects', List.of(p1,p2))
```
